src/main/java/io/vertx/core/file/AsyncFile.java

```diff
@@ -181,4 +181,30 @@
    */
   @Fluent
   AsyncFile setReadBufferSize(int readBufferSize);
+
+  /**
+   * Like {@link #size()} but blocking.
+   *
+   * @throws FileSystemException if an error occurs
+   */
+  default long sizeBlocking() {
+    return -1;
+  }
+
+  /**
+   * Like {@link #size()} but the {@code handler} will be called when the operation is complete or if an error occurs.
+   */
+  default void size(Handler<AsyncResult<Long>> handler) {
+    Future<Long> future = size();
+    if (handler != null) {
+      future.onComplete(handler);
+    }
+  }
+
+  /**
+   * @return the size of the file, or {@code -1} if not supported
+   */
+  default Future<Long> size() {
+    return Future.succeededFuture(-1L);
```

Comment:
does it happen to have unsupported operations for this ?

Reply:
Same answer :smiley:

Seriously, if you both believe I'm over-cautious, I'm fine with removing this.

cc @pmlopes